submissions/src/main/kotlin/gov/cdc/prime/reportstream/submissions/controllers/SubmissionController.kt

```diff
@@ -0,0 +1,185 @@
+            logger.info("Uploaded report to blob storage: blobUrl=${blobClient.blobUrl}")
+
+            // Create the message for the queue
+            val message = mapOf(
+                "reportId" to reportId.toString(),
+                "blobUrl" to blobClient.blobUrl,
+                "headers" to filterHeaders(headers)
+            )
+            val objectMapper = jacksonObjectMapper().registerModule(JavaTimeModule())
+            val messageString = objectMapper.writeValueAsString(message)
+            logger.debug("Created message for queue")
+
+            // Upload to Queue
+            val queueClient = queueServiceClient.getQueueClient(queueName)
+            queueClient.createIfNotExists()
+            queueClient.sendMessage(messageString)
+            logger.info("Sent message to queue: queueName=$queueName")
+
+            // Insert into Table
+            // TableEntity() sets PartitionKey and RowKey. Both are required by azure and combine to create the PK
+            val tableEntity = TableEntity(reportId.toString(), reportId.toString())
+            val tableProperties = mapOf(
+                "report_received_time" to reportReceivedTime.toString(),
+                "report_accepted_time" to reportReceivedTime.toString(), // Will be updated when the report is accepted
+                "report_id" to reportId.toString(),
+                "status" to status
+            )
+            tableClient.createEntity(tableEntity.setProperties(tableProperties))
+            logger.info("Inserted report into table storage: reportId=$reportId")
+
+            // Create and publish custom event
+            val reportReceivedEvent = ReportReceivedEvent(
+                timeStamp = reportReceivedTime,
+                reportId = reportId,
+                parentReportId = reportId,
+                rootReportId = reportId,
+                headers = filterHeaders(headers),
+                sender = clientId!!,
+                senderIP = headers["x-azure-clientip"].toString(),
+                fileSize = headers["content-length"].toString(),
+                blobUrl = blobClient.blobUrl
+            )
+            logger.debug("Created ReportReceivedEvent")
+
+            // Log to Application Insights
+            telemetryClient.trackEvent(
+                "ReportReceivedEvent",
+                mapOf("event" to objectMapper.writeValueAsString(reportReceivedEvent)),
+                null)
+            telemetryClient.flush()
+            logger.info("Tracked ReportReceivedEvent with Application Insights")
+
+            val response =
+                CreationResponse(
+                    reportId,
+                    status,
+                    OffsetDateTime.now(),
+                )
+            logger.info("Report submission successful: reportId=$reportId")
+
+            return ResponseEntity(response, HttpStatus.CREATED)
+        } catch (e: Exception) {
```

Comment:
I think this could be simplified by using a spring `ExceptionHandler`

https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-controller/ann-exceptionhandler.html

Reply:
added an exception handling class. Good call. It's wayyyyyy better